### PR TITLE
Add a parameter register for IS-04 capabilities

### PR DESCRIPTION
**Pull request by garethsb**
Updated #12 to reflect work-in-progress of the AMWA NMOS Receiver Capabilities activity group.

### CODE REVIEW

**Comment by garethsb-sony** [comment]
Discussion during Receiver Capabilities activity group 2-Nov-2020 meeting concluded that it would be useful to have a JSON Schema that allowed IS-04 Receiver resources to be validated as correctly implementing the Receiver capabilities in this register.

**Review by andrewbonney — APPROVED** [review]

**Review by andrewbonney — APPROVED** [review]

File: capabilities/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# NMOS Capabilities
+
+This Capabilities parameter register contains values that may be used to identify a capability, used in the 'caps' property of the resources defined in the [AMWA IS-04 NMOS Discovery and Registration Specification](https://github.com/AMWA-TV/nmos-discovery-registration).
```
Comment (garethsb-sony) [inline capabilities/README.md:3]:
```suggestion
This Capabilities parameter register contains values that may be used to identify a capability, used in the 'caps' property of the resources defined in the [AMWA IS-04 NMOS Discovery and Registration Specification](https://specs.amwa.tv/is-04/).
```

File: capabilities/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+Manufacturers MAY use their own namespaces to indicate capabilities which are not currently defined within the NMOS namespace (`urn:x-nmos:cap:`). In order to avoid collisions with simple names allocated by AMWA specifications, they MUST NOT use capability names that do not start with `urn:`.
+
+Capabilities are most used by IS-04 Receivers to indicate what they may consume, but may be used by other 'caps' objects in the future, potentially to indicate what they may be re-configured to generate.
+
+## Values
+
+- **Name:** media_types
+  - **Description:** Identifies Flows which may be created or consumed based upon their 'media_type' attribute.
+  - **Specification:** [AMWA IS-04 v1.1](https://github.com/AMWA-TV/nmos-discovery-registration/tree/v1.1.x)
```
Comment (garethsb-sony) [inline capabilities/README.md:23]:
```suggestion
  - **Specification:** [AMWA IS-04 v1.1](https://specs.amwa.tv/is-04/v1.1.x/)
```

This should change to .../v1.1/ when that means the latest published patch release on the v1.1 branch, after v1.1 release becomes v1.1.0.

File: capabilities/README.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  - **Applicability:** AMWA IS-04 v1.1+
+- **Name:** event_types
+  - **Description:** Identifies Sources or Flows which may be created or consumed based upon their 'event_type' attribute.
+  - **Specification:** [AMWA IS-04 v1.3](https://github.com/AMWA-TV/nmos-discovery-registration/tree/v1.3-dev)
```
Comment (garethsb-sony) [inline capabilities/README.md:27]:
```suggestion
  - **Specification:** [AMWA IS-04 v1.3](hhttps://specs.amwa.tv/is-04/v1.3.x/)
```

File: capabilities/README.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  - **Applicability:** AMWA IS-04 v1.3+
+- **Name:** constraint_sets
+  - **Description:** Identifies streams from Senders which may be created or consumed based upon the attributes of the associated Source or Flow or contents of the associated transport file.
+  - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)
```
Comment (garethsb-sony) [inline capabilities/README.md:31]:
```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

File: capabilities/README.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  - **Applicability:** AMWA IS-04
+  - **Note:** A JSON schema for Constraint Sets, supporting validation of all the Constraint Set Metadata and Parameter Constraints defined in this register, is available as **[constraint_set.json](constraint_set.json)**.
+    It MAY be used in place of the file with the same name in the AMWA BCP-004-01 specification.
+- **Name:** version
+  - **Description:** String formatted TAI timestamp (<seconds>:<nanoseconds>) indicating when an attribute of the 'caps' object last changed.
+  - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)
```
Comment (garethsb-sony) [inline capabilities/README.md:37]:
```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

File: capabilities/README.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  - **Applicability:** AMWA IS-04
+
+### Constraint Set Metadata
+
+- **Name:** urn:x-nmos:cap:meta:label
+  - **Description:** Freeform string label to provide a human-readable name for a Constraint Set.
+  - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)
```
Comment (garethsb-sony) [inline capabilities/README.md:44]:
```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

File: capabilities/README.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+  - **Applicability:** AMWA IS-04
+- **Name:** urn:x-nmos:cap:meta:preference
+  - **Description:** Expresses the relative 'weight' that the Receiver assigns to its preference for the streams satisfied by the associated Constraint Set.
+  - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)
```
Comment (garethsb-sony) [inline capabilities/README.md:48]:
```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

File: capabilities/README.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  - **Applicability:** AMWA IS-04
+- **Name:** urn:x-nmos:cap:meta:enabled
+  - **Description:** Indicates whether a Constraint Set is available to use immediately (true) or whether this is an offline capability which can be activated via some unspecified configuration mechanism (false).
+  - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)
```
Comment (garethsb-sony) [inline capabilities/README.md:52]:
```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

**Review by garethsb-sony — COMMENTED** [review]
Fix links

Links are now to branches on specs.amwa.tv but should change to links to minor version, i.e. latest patch release, as soon as possible.

**Pull request merged**